File: src/presets/default.ts
import {
  StatusLine, Sep, Git, Group, Branch, Insertions, Deletions,
  Context, Tokens, Percent, Literal,
} from '../dsl/index.js';
import type { SegmentNode } from '../types.js';

export const defaultPreset: SegmentNode[] = StatusLine(() => [
  { type: 'pwd.smart', provider: 'pwd', style: { color: 'cyan' } },
  { type: 'pwd.name', provider: 'pwd', style: { color: 'cyanBright', bold: true } },
  Sep({ char: '|', dim: true }),
  Git()(() => [
    Branch({ color: 'whiteBright', bold: true, icon: '\ue0a0 ' }),
    Group({ prefix: ' [', suffix: ']' })(() => [
      Insertions({ color: 'green', prefix: '+' }),
      Deletions({ color: 'red', prefix: ' -' }),
    ]),
  ]),
  Sep({ char: '|', dim: true }),
  Context()(() => [
    Literal({ text: 'ctx: ' }),
    Tokens({ bold: true }),
    Literal({ text: ' (' }),
    Percent(),
    Literal({ text: ')' }),
  ]),
]);

File: internal/preset/default.go
package preset

import "github.com/jheddings/ccnow/internal/types"

func defaultPreset() []types.SegmentNode {
	return []types.SegmentNode{
		{Type: "pwd.smart", Provider: "pwd", Style: &types.StyleAttrs{Color: "31"}},
		{Type: "pwd.name", Provider: "pwd", Style: &types.StyleAttrs{Color: "39", Bold: true}},
		{
			Type:  "group",
			Style: &types.StyleAttrs{Prefix: " | ", Color: "240"},
			Children: []types.SegmentNode{
				{Type: "git.branch", Provider: "git", Style: &types.StyleAttrs{Color: "whiteBright", Bold: true, Prefix: "\ue0a0 "}},
				{Type: "git.insertions", Provider: "git", Style: &types.StyleAttrs{Color: "green", Prefix: " \u00b7 +"}},
				{Type: "git.deletions", Provider: "git", Style: &types.StyleAttrs{Color: "red", Prefix: " -"}},
			},
		},
		{
			Type:  "group",
			Style: &types.StyleAttrs{Prefix: " | "},
			Children: []types.SegmentNode{
				{Type: "context.tokens", Provider: "context", Style: &types.StyleAttrs{Color: "white", Bold: true}},
				{Type: "context.percent", Provider: "context", Style: &types.StyleAttrs{Color: "white", Prefix: " (", Suffix: ")"}},
			},
		},
		{Type: "session.duration", Provider: "session", Style: &types.StyleAttrs{Color: "magenta", Prefix: " \u00b7 "}},
	}
}
